homeassistant/components/eq3btsmart/climate.py

```diff
@@ -20,79 +24,312 @@
+            [PRECISION_TENTHS, PRECISION_HALVES, PRECISION_WHOLE]
+        ),
+        vol.Optional(CONF_COLD_TOLERANCE, default=DEFAULT_TOLERANCE): vol.Coerce(float),
+        vol.Optional(CONF_HOT_TOLERANCE, default=DEFAULT_TOLERANCE): vol.Coerce(float),
+    }
+)
 
 PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
     {vol.Required(CONF_DEVICES): vol.Schema({cv.string: DEVICE_SCHEMA})}
 )
 
 SUPPORT_FLAGS = SUPPORT_TARGET_TEMPERATURE | SUPPORT_PRESET_MODE
+SUPPORT_FLAGS_SENSOR_CONTROL = SUPPORT_TARGET_TEMPERATURE
 
 
-def setup_platform(hass, config, add_entities, discovery_info=None):
+async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
     """Set up the eQ-3 BLE thermostats."""
     devices = []
 
     for name, device_cfg in config[CONF_DEVICES].items():
         mac = device_cfg[CONF_MAC]
-        devices.append(EQ3BTSmartThermostat(mac, name))
-
-    add_entities(devices, True)
-
-
-class EQ3BTSmartThermostat(ClimateEntity):
+        sensor_entity_id = device_cfg.get(CONF_SENSOR)
+        sensor_control = device_cfg.get(CONF_SENSOR_CONTROL)
+        min_temp = device_cfg.get(CONF_MIN_TEMP)
+        max_temp = device_cfg.get(CONF_MAX_TEMP)
+        precision = device_cfg.get(CONF_PRECISION)
+        cold_tolerance = device_cfg.get(CONF_COLD_TOLERANCE)
+        hot_tolerance = device_cfg.get(CONF_HOT_TOLERANCE)
+
+        devices.append(
+            EQ3BTSmartThermostat(
+                mac,
+                name,
+                sensor_entity_id,
+                sensor_control,
+                min_temp,
+                max_temp,
+                precision,
+                cold_tolerance,
+                hot_tolerance,
+            )
+        )
+
+    async_add_entities(devices, True)
+
+
+class EQ3BTSmartThermostat(ClimateEntity, RestoreEntity):
     """Representation of an eQ-3 Bluetooth Smart thermostat."""
 
-    def __init__(self, _mac, _name):
+    def __init__(
+        self,
+        _mac,
+        _name,
+        _sensor_entity_id,
+        _sensor_control,
+        _min_temp,
+        _max_temp,
+        _precision,
+        _cold_tolerance,
+        _hot_tolerance,
+    ):
         """Initialize the thermostat."""
         # We want to avoid name clash with this module.
         self._name = _name
         self._thermostat = eq3.Thermostat(_mac)
 
+        self._sensor_entity_id = _sensor_entity_id
+        self._sensor_control = _sensor_control
+        self._min_temp = _min_temp
+        self._max_temp = _max_temp
+        self._temp_precision = _precision
+        self._cold_tolerance = _cold_tolerance
+        self._hot_tolerance = _hot_tolerance
+        self._cur_temp = None
+        self._target_temp = None
+        self._hvac_mode = None
+        self._active = False
+        self._temp_lock = asyncio.Lock()
+
+    async def async_added_to_hass(self):
+        """Run when entity about to be added."""
+        await super().async_added_to_hass()
```

Comment:
Is this necessary?

Reply:
Used the same approach as [generic_thermostat](https://github.com/home-assistant/core/blob/b2c9bd2ca67df277093790e3fc0def34d9f7fb04/homeassistant/components/generic_thermostat/climate.py#L186).
If it's there, I assumed it's needed.